### ISSUE TITLE
Zowe CLI V2 - Need a technical note / whitepaper for the implementation of the profiles

### ISSUE THREAD

**Issue by FALLAI-Denis** [zowe/zowe-cli#1407]
Hi,

The move to Zowe CLI V2 completely changes profile management.

The online documentation does not seem very clear to me...

The online help, direct or in web display, does not provide any explanation...

It seems to me **ESSENTIAL** to have a reference document to explain how profiles work in V2, with all the use cases listed.

Moreover, the absence of a native Zowe CLI command to manage profiles, and the obligation to directly edit the configuration file, is **UNNACEPTABLE** for us, and we will block the Zowe CLI V1 to Zowe CLI V2 migration in our company. .

**Comment by FALLAI-Denis** [comment]
Zowe CLI V2 profiles management:

![image](https://user-images.githubusercontent.com/63861965/167094603-195b41a2-6a98-4837-bb7a-332b6cbd612f.png)

**Comment by FALLAI-Denis** [comment]
#1406 #1405 #1404 #1403

**Comment by VitGottwald** [comment]
I agree with @FALLAI-Denis 's request for a white paper or a reference manual describing in detail the mechanism how v2 profiles work.

I worked with a customer today who was unable to get a successful setup. The documentation did not help him to understand what kind of config file he needed and what was the difference between them.

We spent 1/2 of our working day analysing the problem with his setup. Eventually it turned out that at some point he most likely  ran `zowe config init` in his home folder. This created zowe.config.json there that was subsequently used when he ran zowe commands or used ZOWE Explorer (he had a project that resided in a folder in his home directory open). The most confusing part was that we did not see `zowe.config.json` in the project but running `zowe config secure --global-config` did not help with setting credentials. What finally helped us was right clicking on the profile in zowe explorer to "Update Profile" which revealed the config in his home folder.

At the end when we got it working he asked for a white paper with all the details how v2 profiles work AND for a short video showing 

- first time setup happy path
- upgrade from v1 happy path

**Comment by gejohnston** [comment]
Do we consider this issue resolved by this recent Zowe user documentation:

[Understanding profile configurations in Zowe CLI](https://docs.zowe.org/stable/user-guide/cli-using-understand-profiles-configs)

and this engineering whitepaper:

[How Zowe CLI team configuration files are merged together](https://github.com/zowe/zowe-cli/blob/master/docs/How_config_files_are_merged.md)

**Comment by FALLAI-Denis** [comment]
Hi,

Most of the problem is due to the absence of commands to declare or update the profiles.
It is not acceptable to have to manually edit a file whose structure is complex and for which all input errors to completely block the operation of Zowe CLI.
Something simple should be simple to explain. Obviously this is not the case on reading the documents given in reference.

For me the situation is still the same: it's a pain to have to manage the profiles since the switch to V2.

**Comment by FALLAI-Denis** [comment]
Hi,

No change with Zowe V3…
🤨

**Comment by zFernand0** [comment]
This enhancement has had no community activity for 12 months. The issue also has less than 5 up-votes by the community. No action on this enhancement is targeted for the next 2 calendar quarters. Therefore, this enhancement is being closed. If you feel that this enhancement should continue to be available for community up-votes, you may reopen this issue.
